python/PiFinder/composite_object.py

```diff
@@ -2,10 +2,20 @@
 from dataclasses import dataclass, field
 import numpy as np
 import json
-from PiFinder.utils import is_number
 from typing import List
 
 
+def is_number(s):
```

Comment:
I'm not seeing where this is used, but it seems like the same number of lines could be used to just ask forgiveness rather than permission inline
```
if is_number(x):
    f=float(x)
    # do the float thing with f
else:
    # do the non float thing
```
vs
```
try:
    f=float(x)
    # do the float thing
except (ValueError, TypeError):
    # do the non float thing
```

Using the function might be nice if there ends up being a lot of stuff going on in the 'float stuff here' section which could also throw a ValueError.  

If you do want to keep this function, no need for the s is None check if you catch both Value and Type errors :+1:

Reply:
can be used inline:
    def _filter_floats(self):
        return [float(x) for x in self.mags if is_number(x)]
        
agreed with the None